x-pack/plugins/security/server/__fixtures__/request.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import url from 'url';
+import { KibanaRequest } from '../../../../../src/core/server';
+
+interface RequestFixtureOptions {
+  headers?: Record<string, string>;
+  params?: Record<string, unknown>;
+  path?: string;
+  search?: string;
+  payload?: unknown;
+}
+
+export function requestFixture({
+  headers = { accept: 'something/html' },
+  params,
+  path = '/wat',
+  search = '',
+  payload,
+}: RequestFixtureOptions = {}) {
+  return KibanaRequest.from({
```

Comment:
Yes, I mentioned that in the issue description, that'd be awesome. The more "universal" (so that people can tune them as needed without depending on the platform team to extend the mock and hence resorting to the "invent-the-wheel" things) mocks core can provide, the better.

Reply:
added https://github.com/elastic/kibana/issues/40186